scripts/zones/AlTaieu/globals.lua

```diff
@@ -0,0 +1,81 @@
+-- Zone: Al'Taieu (33)
+-- Desc: this file contains functions that are shared by multiple luas in this zone's directory
+-----------------------------------
+local ID = require("scripts/zones/AlTaieu/IDs")
+require("scripts/globals/missions")
+require("scripts/globals/npc_util")
+-----------------------------------
+
+local antiquityVars =
+{
+    -- [crystalOffset] = {thisFightDone, thisCsAcquired, otherCsAcquired1, otherCsAcquired2, firstCsParam, secondCsParam1, secondCsParam2}
+    [0] = {'[SEA][AlTieu]SouthTower', '[SEA][AlTieu]SouthTowerCS', '[SEA][AlTieu]EastTowerCS', '[SEA][AlTieu]WestTowerCS', 0, 2, 1},
+    [1] = {'[SEA][AlTieu]WestTower', '[SEA][AlTieu]WestTowerCS', '[SEA][AlTieu]SouthTowerCS', '[SEA][AlTieu]EastTowerCS', 2, 1, 0},
+    [2] = {'[SEA][AlTieu]EastTower', '[SEA][AlTieu]EastTowerCS', '[SEA][AlTieu]SouthTowerCS', '[SEA][AlTieu]WestTowerCS', 1, 2, 0},
+}
+
+return {
+    --[[..............................................................................................
+        Garden of Antiquity: player clicks a rubious crystal tower
+
+        Event 161 first post-fight CS, mentions current tower by param: 0 = South, 1 = East, 2 = West
+        Event 162 second post-fight CS, mentions final tower by param: 0 = South, 1 = East, 2 = West
+        Event 163 third post-fight CS
+        ..............................................................................................]]
+    rubiousCrystalOnTrigger = function(player, npc)
+        local npcId = npc:getID()
+        local crystalOffset = npcId - ID.npc.RUBIOUS_CRYSTAL_BASE
+        local ruaernOffset = ID.mob.RUAERN_BASE + crystalOffset * 3
+        local cop = player:getCurrentMission(COP)
+        local copStat = player:getCharVar("PromathiaStatus")
+        local cVar = antiquityVars[crystalOffset]
+        local thisFightDone = player:getCharVar(cVar[1]) == 1
+        local thisCsAcquired = player:getCharVar(cVar[2]) == 1
+
+        -- spawn ru'aerns
+        if
+            cop == tpz.mission.id.cop.GARDEN_OF_ANTIQUITY and
+            copStat < 3 and
```

Comment:
I think that here we should leave the == 2 because we should not be able to pops if the CS before has not been done. When we start this mission we have Promathia status to 0 and after the CS we set promathia status to 2. no?

Reply:
The wiki mentions that you can do the fight (and be credited for it) before getting the cutscene at the crystalline field. 
https://ffxiclopedia.fandom.com/wiki/Garden_of_Antiquity#Southern_Tower_.28H-13.29

Reply:
it's ok for me

Reply:
thanks for this contribs 👍